buildSrc/src/main/groovy/local.java.container-conventions.gradle

```diff
@@ -36,63 +38,86 @@ dockerCompose {
 }
 
 def dockerStart = tasks.register('dockerStart', DefaultTask) {
-    dependsOn "dockerPrune", "docker", "dockerRun"
-}
-
-def dockerPrune = tasks.register('dockerPrune', DefaultTask) {
-    mustRunAfter 'dockerStop', 'dockerRemoveContainer'
-    dependsOn 'dockerPruneContainer', 'dockerPruneImage'
+    group = "Docker Run"
+    description = "Remove unused containers, build image, run container"
+    dependsOn tasks.named("dockerPrune")
+    dependsOn tasks.named("docker")
+    dependsOn tasks.named("dockerRun")
 }
 
 def dockerPruneContainer = tasks.register('dockerPruneContainer', Exec) {
+    group = "Docker"
+    description = "Remove unused containers"
     executable "docker"
     args "container", "prune", "-f"
 }
 
 def dockerPruneImage = tasks.register('dockerPruneImage', Exec) {
+    group = "Docker"
+    description = "Remove unused images"
     executable "docker"
     args "image", "prune", "-f"
 }
 
 def dockerPruneVolume = tasks.register('dockerPruneVolume', Exec) {
+    group = "Docker"
+    description = "Remove unused volumes"
     executable "docker"
     args "volume", "prune", "-f"
 }
 
+def dockerPrune = tasks.register('dockerPrune', DefaultTask) {
```

Comment:
One thing I can't figure out is how these functions are called and from which part of the build.

Reply:
It doesn't seem like `dockerPrune` in particular is being called as part of a build. You'd have to call `dockerStop` or `dockerRemoveContainer` (lines 72-73). To explore, try `./gradlew tiOrder dcPrune` or `./gradlew tiTree dcPrune`.